src/librustc/middle/infer/error_reporting.rs

```diff
@@ -479,6 +479,14 @@ impl<'a, 'tcx> ErrorReporting<'tcx> for InferCtxt<'a, 'tcx> {
                          trace: TypeTrace<'tcx>,
                          terr: &TypeError<'tcx>)
                          -> DiagnosticBuilder<'tcx> {
+        fn is_simple_type<'tcx>(ty: Ty<'tcx>) -> bool {
```

Comment:
This should be a method on `TyS` rather than a function here - `is_primitive` maybe